hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -85,17 +84,17 @@ public HoodieHiveClient(HiveSyncConfig cfg, HiveConf configuration, FileSystem f
       } else {
         ddlExecutor = cfg.useJdbc ? new JDBCExecutor(cfg, fs) : new HiveQueryDDLExecutor(cfg, fs, configuration);
       }
-      this.client = Hive.get(configuration).getMSC();
+      this.client = HiveMetastoreClientFactory.create(configuration, cfg.hiveVersion == null ? HiveShimLoader.getHiveVersion() : cfg.hiveVersion);
```

Comment:
can we assign this `HiveShimLoader.getHiveVersion()` as default? so that we don't need to check every time